lib/qesapdeployment.pm

```diff
@@ -600,22 +601,24 @@ sub qesap_execute_conditional_retry {
 
     while ($args{retries} > 0) {
         if ($ret[0]) {
-            if (qesap_file_find_string(file => $ret[1], search_string => $args{error_string})) {
-                record_info('DETECTED ' . uc($args{cmd}) . ' ERROR', $args{error_string});
+            if (qesap_file_find_strings(file => $ret[1], search_strings => $args{error_list})) {
+                record_info('DETECTED ' . uc($args{cmd}) . ' ERROR', $args{error_list});
 
                 # Executing terraform destroy before retrying terraform apply
                 if ($args{destroy_terraform}) {
-                    qesap_execute(
+                    my @destroy_ret = qesap_execute(
                         cmd => 'terraform',
                         cmd_options => '-d',
                         logname => "qesap_exec_terraform_destroy_before_retry$args{retries}.log.txt",
                         verbose => 1,
                         timeout => 1200);
+                    die "Terraform destroy exited with non-zero code." unless ($destroy_ret[0] == 0);
```

Comment:
Not something I suggest to change in this PR but something I like we discuss in the future.

I think to remember that:
- this function `qesap_execute_conditional_retry` has been designed as a wrapper around `qesap_execute`
- `qesap_execute` is designed not to have any die, but always to return something. Idea is delegate to the caller (in the test module) how to handle a failure (eventually die).

We violate this design principle in this wrapper
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18639/files#diff-23c988d39d71dc252bb5dce95a699047e93941078f2152ab4a6587286e9a4e47R548

Maybe one day we can try to apply the "no die only return" principle in this wrapper too (not now).